src/ci/tested_versions_utils.py

```diff
@@ -1,38 +1,180 @@
 from __future__ import annotations
 
 import os
+import re
 from contextlib import contextmanager
 from dataclasses import dataclass
-from typing import Tuple
+from typing import List, Union
+
+# Major, minor, patch and (non semver standard, suffix)
+_semanticVersionPattern = re.compile(r'(!)?(\d+).(\d+).(\d+)([^\s]*)(?:\s*#\s*(.*))?')
+_splitVersionFromCommentPattern = re.compile(r'(!)?([^\s]*)(?:\s*#\s*(.*))?')
+
+# This file implements a custom version parsing and sorting mechanism,
+# as `packaging.version` has strange behaviors that won't work for other
+# languages, like:
+#
+# ```
+# >>> from packaging.version import parse
+# >>> str(parse("1.2.4c"))
+# '1.2.4rc0'
+# ```
+#
+# ```
+# >>> from packaging.version import parse
+# >>> str(parse("1.2.4b"))
+# '1.2.4b0'
+# ```
+
+
+@dataclass(frozen=True)
+class NonSemanticVersion:
+    supported: bool
+    version: str
+    comment: str
+
+    def __eq__(self, other):
+        if isinstance(other, SemanticVersion):
+           return False
+
+        return self.version == other.version
+
+    def __lt__(self, other):
+        if isinstance(other, SemanticVersion):
+           return True
+
+        return self.version < other.version
+
+
+@dataclass(frozen=True)
+class SemanticVersion:
+    supported: bool
+    major: int
+    minor: int
+    patch: int
+    suffix: str
+    comment: str
+
+    @property
+    def version(self):
+        return f"{self.major}.{self.minor}.{self.patch}{self.suffix or ''}"
+
+    def __eq__(self, other):
+        if isinstance(other, NonSemanticVersion):
```

Comment:
```suggestion
        if not isinstance(other, SemanticVersion):
```

Reply:
Done in dfb0576a9213e84b59b2b3970f5c8738923abc7a